executor/src/wasm_executor.rs

```diff
@@ -96,10 +97,13 @@ impl_function_executor!(this: FunctionExecutor<'e, E>,
 	ext_memcmp(s1: *const u8, s2: *const u8, n: usize) -> i32 => {
 		if let (Ok(sl1), Ok(sl2))
 			= (this.memory.get(s1, n as usize), this.memory.get(s2, n as usize)) {
-			use memcmp::Memcmp;
-			(&sl1).memcmp(&sl2) as i32
+			unsafe {
+				memcmp(sl1.as_ptr() as *const u8 as *const c_void,
```

Comment:
can be replaced with safe code:
```rust
match sl1.cmp(&sl2) {
    Ordering::Greater => 1,
    Ordering::Less => -1,
    Ordering::Equal => 0,
}
```

Reply:
https://doc.rust-lang.org/stable/src/core/slice/mod.rs.html#2571

it's specialized to compile down to a `memcmp` and be inlined.

Reply:
oh nice.